inductiva/_benchmarks/benchmark.py

```diff
@@ -0,0 +1,119 @@
+"""API for Benchmarking"""
+import json
+from typing import Self
+from inductiva import types
+from inductiva.simulators.simulator import Simulator
+from inductiva.projects.project import Project
+from inductiva.resources import machine_groups
+
+
+class Benchmark(Project):
+    """Represents the benchmark runner."""
+
+    def __init__(self, name: str, append: bool = True):
+        """
+        Initializes a new Benchmark instance.
+
+        Args:
+            name (str): The name of the benchmark runner.
+            append (bool): Indicates whether to allow adding runs to the 
+            existing benchmark (default is True).
+        """
+        super().__init__(name=name, append=append)
+        self.runs = []
+
+    def add_run(
+        self,
+        simulator: Simulator,
+        input_dir: str,
+        on: types.ComputationalResources,
+        **kwargs,
+    ) -> Self:
+        """
+        Adds a simulation run to the benchmark.
+
+        Args:
+            simulator (Simulator): The simulator to be used for this run.
+            input_dir (str): The directory containing input files for the
+            simulation.
+            on (types.ComputationalResources): The computational resources to
+            run the simulation on.
+            **kwargs: Additional keyword arguments for the simulator run.
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        self.runs.append((simulator, input_dir, on, kwargs))
+        return self
+
+    def run(self, num_repeats: int = 2) -> Self:
+        """
+        Executes all added runs.
+
+        Each run is executed the specified number of times, and the collection
+        of runs is cleared afterwards.
+
+        Args:
+            num_repeats (int): The number of times to repeat each simulation
+            run (default is 2).
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        with self:
+            for simulator, input_dir, machine_group, kwargs in self.runs:
+                machine_group.start(wait_on_pending_quota=True)
```

Comment:
Allow the configuration of wait_on_pending_quota on the Benchmark constructor. You can make it True by default.

Reply:
I think we should keep this version of the Benchmark API as simple as possible to highlight only the most important parameters needed to create, configure, run, and export a benchmark.

We can consider including additional parameters in future versions, for example those that could also be passed to the ```wait``` method.

What do you think?